src/Eloquent/Builder.php

```diff
@@ -203,55 +199,42 @@ public function raw($value = null)
     }
 
     /**
-     * Attempt to create the record if it does not exist with the matching attributes.
-     * If the record exists, it will be returned.
+     * Get the first record matching the attributes. If the record is not found, create it.
      *
-     * @param  array $attributes The attributes to check for duplicate records
-     * @param  array $values     The attributes to insert if no matching record is found
+     * @param  array $attributes
+     * @param  array $values
      */
-    public function createOrFirst(array $attributes = [], array $values = []): Model
+    public function firstOrCreate(array $attributes = [], array $values = []): Model|Builder
     {
-        if ($attributes === [] || $attributes === ['_id' => null]) {
-            throw new InvalidArgumentException('You must provide attributes to check for duplicates. Got ' . json_encode($attributes));
+        $instance = (clone $this)->where($attributes)->first();
+        if ($instance !== null) {
+            return $instance;
         }
 
-        // Apply casting and default values to the attributes
-        // In case of duplicate key between the attributes and the values, the values have priority
-        $instance = $this->newModelInstance($values + $attributes);
+        // createOrFirst is not supported in transaction.
+        if ($this->getConnection()->getSession()?->isInTransaction()) {
+            return $this->create(array_merge($attributes, $values));
```

Comment:
Should `+` be used instead of `array_merge` to ensure that string _and_ numeric keys overwrite each other instead of possibly appending numerics?

Reply:
Copied from https://github.com/laravel/framework/blob/97238d27e271d6f8b36b64680756f788130f19c5/src/Illuminate/Database/Eloquent/Builder.php#L592